src/plugins/discover/public/application/main/components/layout/discover_layout.tsx

```diff
@@ -256,6 +256,21 @@ export function DiscoverLayout({ stateContainer }: DiscoverLayoutProps) {
 
   const onFilter = isEsqlMode ? onPopulateWhereClause : onAddFilter;
 
+  const canSetBreakdownField = useMemo(
+    () =>
+      isOfAggregateQueryType(query)
+        ? dataView?.isTimeBased() && !hasTransformationalCommand(query.esql)
+        : true,
+    [dataView, query]
+  );
+
+  const onSetBreakdownField = useCallback(
```

Comment:
Since `onAddBreakdownField` is used in all other files, let's rename here too then

```suggestion
  const onAddBreakdownField = useCallback(
```